hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+
+Early on during this stage, the various secret unlock tokens are also programmed into the secret partition and harvested by the silicon creator.
+
+### Provisioning
+
+During this stage, the device is provisioned with the final firmware and a "unique" seed or identity.
+The secret partition is populated with root secrets and keys that are critical to establishing the device identity.
+
+As part of injecting the final firmware, the SKU-specific hardware and software configurations are also programmed.
+
+### Life Cycle Partition
+
+The life cycle partition is active throughout all stages and hence it is the **ONLY** partition that cannot be locked.
+After the device finishes provisioning and goes into production, it must retain the ability to transition back to RMA in case of unexpected failures.
+
+In order to support this transition, the life-cycle state and counters must always be update-able (**TODO: add link to life-cycle docs for more info**).
+
+## Locking a Partition
+
+Write access to a partition can be permanently locked when software determines it will no longer make any updates to that partition.
+To lock, a truncated integrity constant (32bit) is calculated and programmed alongside the other data of that partition.
+
+Once the "integrity digest" is non-zero, no further updates are allowed.
+If the partition is secret, software is in addition no longer able to read the contents (see secret partiton description further below).
+
+Note however, in all partitions, the digest itself is **ALWAYS** readable.
+This gives software an opportunity to confirm that the locking operation has proceeded correctly, and if not, scrap the part immediately.
+
+Calculation of the integrity digest depends on whether the partition requires periodic background verification.
+
+### Software Configuration Partition
+
+The software configuration partition uses the OTP as non-volatile flags and storage.
+As such, the contents of this partition are usually consumed once as part of code execution, or moved to another storage compartment somewhere in the design (for example, the clock calibration values are programmed to the clocking module, and the LDO calibration values are programmed to the regulator wrapper).
+
+As such, it is not necessary to check periodically at the OTP source.
+Instead, software can simply check as part of secure boot and take other measures when these values are programmed into peripherals.
+
+For this partition it is thus the responsibility of software to calculate the integrity digest and program it into the OTP.
+It is also reasonable to shadow (parts) of this partition in main memory, and there is not an immediate impact from OTP contents to hardware.
+
+### Hardware Configuration and Secret Partitions
+
+The hardware and secret partitions directly affect downstream hardware.
+The contents must go through periodic integrity checks and so the stored digest is calculated by hardware when software provides the intent to lock.
```

Comment:
The contents must go through periodic integrity checks and (so-> redundant?) the stored digest is calculated by hardware when software provides the intent to lock.

Reply:
hm... "so" should be a "hence" or a "therefore" in this case. I want to point out that the periodic hw checks are the reason this digest needs to be calculated in hw (as opposed to sw as for the sw partitions).